src/CFamily/CMake/MacroEvaluationService.cs

```diff
@@ -34,17 +40,62 @@ internal interface IMacroEvaluationService
 
     internal class MacroEvaluationService : IMacroEvaluationService
     {
+        private readonly ILogger logger;
+        private readonly IMacroEvaluator macroEvaluator;
+
+        // Looks for patterns like "${name}" and "${prefix.name}" and captures the [prefix, name] tuple.
+        private static readonly Regex CMakeSettingsMacroRegex = new Regex(
+            "\\${" +                 // Match opening literals "${". $ is a special Regex symbol that needs be escaped,
+                                     // and "\" needs to be escaped in the C# string.
+            "((?<prefix>\\w+)\\.)" + // One or more word chars followed by ".". Like "$" above, the "." needs to be escaped with "\\".
+                                     // The word chars are captured in a group called "prefix".
+            "?" +                    // Match the previous group 0 or 1 times i.e. the prefix is optional.
+            "(?<name>\\w+)" +        // Capture one or more word consecutive word chars in a group called "name".
+            "}");                    // Match the closing literal.
+
+        public MacroEvaluationService(ILogger logger)
+            :this(logger, new MacroEvaluator()) {}
+
+        internal MacroEvaluationService(ILogger logger, IMacroEvaluator macroEvaluator)
+        {
+            this.logger = logger;
+            this.macroEvaluator = macroEvaluator;
+        }
+
         public string Evaluate(string input, string activeConfiguration, string workspaceRootDir)
         {
             if (input == null)
             {
                 return null;
             }
+
+            var context = new EvaluationContext(activeConfiguration, workspaceRootDir);
+            var sb = new StringBuilder(input);
+
+            foreach(Match match in CMakeSettingsMacroRegex.Matches(input))
+            {
+                var prefix= match.Groups["prefix"].Value; // will be String.Empty if not found
+                var name = match.Groups["name"].Value;
+
+                var evaluatedProperty = macroEvaluator.TryEvaluate(prefix, name, context);
+
+                if (evaluatedProperty == null)
+                {
+                    // Give up if we failed to evaluate any property
+                    LogDebug($"Failed to evaluate macro: {match.Value}");
```

Comment:
VS shows this information to the user, so maybe it should be an info message instead of debug.

Reply:
I wondered about that. At the moment, we'll display a message at a higher level saying we can't evaluate the build root property, which includes the pre-evaluation version of the string i.e. with the macros included, but it won't say which macro we couldn't process.
I was worried that displaying a lower-level message would be too noisy, but it will make it easier for the user to fix any issues. I'll change it and update the test.